CRFModuleValidation/CRFCoreMotionRecorder.swift

```diff
@@ -0,0 +1,275 @@
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+import CoreMotion
+import ResearchSuite
+
+public enum CRFCoreMotionRecorderType : String, Codable {
+    case accelerometer
+    case deviceMotion
+}
+
+public struct CRFCoreMotionRecorderConfiguration : RSDRecorderConfiguration, RSDAsyncActionControllerVendor, Codable {
+    public let identifier: String
+    public let recorderType: CRFCoreMotionRecorderType
+    public let startStepIdentifier: String?
+    public let stopStepIdentifier: String?
+    public let requiresBackgroundAudio: Bool
+    public let frequency: Double?
+    
+    public init(recorderType: CRFCoreMotionRecorderType) {
+        self.init(identifier: recorderType.rawValue, recorderType: recorderType)
+    }
+    
+    public init(identifier: String, recorderType: CRFCoreMotionRecorderType, startStepIdentifier: String? = nil, stopStepIdentifier: String? = nil, requiresBackgroundAudio: Bool = true, frequency: Double? = nil) {
+        self.identifier = identifier
+        self.recorderType = recorderType
+        self.startStepIdentifier = startStepIdentifier
+        self.stopStepIdentifier = stopStepIdentifier
+        self.requiresBackgroundAudio = requiresBackgroundAudio
+        self.frequency = frequency
+    }
+    
+    public var permissions: [RSDPermissionType] {
+        return [RSDStandardPermissionType.coremotion]
+    }
+
+    public func validate() throws {
+        try RSDStandardPermissionType.coremotion.validate()
+    }
+    
+    public func instantiateController(with taskPath: RSDTaskPath) -> RSDAsyncActionController? {
+        switch recorderType {
+        case .accelerometer:
```

Comment:
Should double-check with Larsson that when they just ask for the raw "accelerometer" data they don't also really want raw gyroscope samples too. IIRC ResearchKit includes at least all three streams (accel/gyro/device motion) and possibly a raw magnetometer stream as well.

Reply:
Thanks. This was pulled from ResearchKit directly without much thought. (late night :/)

But those are fairly easy changes.

Also, data from RK is saved to two files with these keys using two managers.